python/runtime/mlir/py_register_dialects.cpp

```diff
@@ -292,10 +259,9 @@ void registerCCDialectAndTypes(py::module &m) {
           });
 }
 
-void bindRegisterDialects(py::module &mod) {
-  registerQuakeDialectAndTypes(mod);
-  registerCCDialectAndTypes(mod);
+static bool registered = false;
```

Comment:
This looks like it was pre-existing, but I'm not a fan of a global like this. Is there no good way to query whether things have been registered already by asking the dictionary of registered stuff?

Reply:
Yes, there are actually two options: (1) makes so the dialects get registered upon Python's module loading or (2) make MLIR python automagically register things.

Option (2) requires a slightly different sort of change that doesn't fit well with the rest of this PR. I had (2) implemented before, but I reverted the change because the CI is failing and I can't reproduce the issue locally, even when using the same docker image :( 

After various experiments of changing the code, adding global state back, I still can't make it pass CI. (So this is the state of my last try, I will update the branch and revisit the change.)